cli/src/lib.rs

```diff
@@ -1950,10 +1952,48 @@ fn validator_flags(cfg: &WithPath<Config>) -> Result<Vec<String>> {
                         flags.push(entry["filename"].as_str().unwrap().to_string());
                     }
                 } else if key == "clone" {
+                    // Client for fetching accounts data
+                    let client = if let Some(url) = entries["url"].as_str() {
+                        RpcClient::new(url.to_string())
+                    } else {
+                        return Err(anyhow!(
+                    "Validator url for Solana's JSON RPC should be provided in order to clone accounts"
+                ));
+                    };
+
                     for entry in value.as_array().unwrap() {
                         // Push the clone flag for each array entry
                         flags.push("--clone".to_string());
                         flags.push(entry["address"].as_str().unwrap().to_string());
+
+                        // Check it is a program account
+                        let pubkey = Pubkey::from_str(flags.last().unwrap())
+                            .map_err(|_| anyhow!("Invalid pubkey {}", flags.last().unwrap()))?;
+
+                        let account = client
+                            .get_account_with_commitment(&pubkey, CommitmentConfig::default())?
```

Comment:
We should use getMultipleAccounts instead. WIth a lot of programs, this can easily get rate limited.

Reply:
ok, i've changed the logic a little bit to make a single request